core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/metrics/NodeMetricHolder.java

```diff
@@ -79,6 +85,11 @@ protected Ratio getRatio()
                         .map(Map.Entry::getKey)
                         .collect(Collectors.toList());
 
+                if (repairedTables.isEmpty())
```

Comment:
The hole method could be replaced with a single stream:

protected Ratio getRatio()
{
   return myTableRepairRatio.entrySet().stream()
            .filter(e -> e.getValue() > 0)
            .map(Map.Entry::getKey)
            .map(tableReference -> tableStorageStates.getDataSize(tableReference) * myTableRepairRatio.get(tableReference))
            .reduce(Double::sum)
            .map(size -> Ratio.of(size, tableStorageStates.getDataSize()))
            .orElse(RATIO_ONE);
}

But maybe that is "too much" ;-/
The NaN is solved at the end when the Optional sum is mapped into a Ratio OR the default value 1...